code/game/objects/items/weapons/RCD.dm

```diff
@@ -597,7 +619,18 @@
  * * amount - an amount of matter to check for
  */
 /obj/item/rcd/proc/checkResource(amount, mob/user)
-	return matter >= amount
+	. = matter >= amount
+	if(!. && user)
+		to_chat(user, no_ammo_message)
+		flick("[icon_state]_empty", src)
+
+	return .
```

Comment:
you don't need to specify the `.` here, IIRC the default return value is `.`

Reply:
didn't know it was literally the same, lmao